### PR TITLE
feat(geth/trace): add field log.position

### PR DESCRIPTION
**Pull request by jsvisa**
## Motivation

Add a field `log.position` in the response of debug_traceXXX RPC with `withLog:true` config(which was added in https://github.com/ethereum/go-ethereum/pull/28389)

ref https://github.com/paradigmxyz/reth/issues/10258

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/rpc-types-trace/src/geth/call.rs [hunk 1]
```diff
@@ -56,6 +56,9 @@ pub struct CallLogFrame {
     /// The data of the log.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub data: Option<Bytes>,
+    /// The position of the log relative to subcalls within the same trace.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
```
Comment (DaniPopes) [inline crates/rpc-types-trace/src/geth/call.rs:60]: pls use alloy_serde::quantity::opt with u64 instead of ruint

**Review by DaniPopes — CHANGES_REQUESTED** [review]

File: crates/rpc-types-trace/src/geth/call.rs (hunk 1 above)
Comment (jsvisa) [reply crates/rpc-types-trace/src/geth/call.rs]: fixed, ptal

**Review by onbjerg — APPROVED** [review]

File: crates/rpc-types-trace/src/geth/call.rs [hunk 2]
```diff
@@ -56,6 +56,9 @@ pub struct CallLogFrame {
     /// The data of the log.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub data: Option<Bytes>,
+    /// The position of the log relative to subcalls within the same trace.
+    #[serde(default, with = "alloy_serde::quantity::opt")]
+    pub position: Option<u64>,
```
Comment (onbjerg) [inline crates/rpc-types-trace/src/geth/call.rs:61]: should we skip serializing this if `None`?

Comment (DaniPopes) [reply crates/rpc-types-trace/src/geth/call.rs:61]: we should because null is generally invalid for this type of stuff

Comment (jsvisa) [reply crates/rpc-types-trace/src/geth/call.rs:61]: Thanks for it, so we should also add skip_if_none

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**